lib/std/json.zig

```diff
@@ -1383,8 +1383,60 @@ pub const ParseOptions = struct {
         Error,
         UseLast,
     } = .Error,
+
+    /// If false, finding an unknown field returns an error.
+    ignore_unknown_fields: bool = false,
 };
 
+fn skipValue(tokens: *TokenStream) !void {
+    const original_depth = tokens.parser.stack_used;
+
+    // Return an error if no value is found
+    _ = try tokens.next();
```

Comment:
you also need the `== original_depth` check here

Reply:
That is indeed an error I failed to see, but that's not what's causing the tests to fail.

[This is the problematic test](https://github.com/ziglang/zig/pull/7918/files#diff-defc5ed2d34b24434f31f2d92e270f4f98a1a32ec07147799eb12c1242d5b0a4R1947). According to my probing, `stack_used` is reduced on the third number's token, and not on the `ArrayEnd`; so, when `skipValue` returns, the parser finds an unexpected token. `complex` and `veryComplex` don't seem to be affected by this, even after adding fields with numbers in them.

Reply:
I couldn't figure out why `stack_used` was being reduced in such a way, so I made a workaround for it, would this be acceptable?

Reply:
Ah! That'll be because of the 1 token lookahead for numbers in the streaming parser.

Does something like this work?:
```diff
diff --git a/lib/std/json.zig b/lib/std/json.zig
index ff258c7f3..e1f7e5df6 100644
--- a/lib/std/json.zig
+++ b/lib/std/json.zig
@@ -1105,6 +1105,10 @@ pub const TokenStream = struct {
         };
     }
 
+    fn stack_used(self: TokenStream) u8 {
+        return self.parser.stack_used + if (self.token != null) 1 else 0;
+    }
+
     pub fn next(self: *TokenStream) Error!?Token {
         if (self.token) |token| {
             self.token = null;

```

Reply:
Shouldn't that be `stackUsed` (instead of `stack_used`)?